Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/FindPeakAutomaticTest.py

```diff
@@ -146,6 +161,38 @@ def test_algorithm_creates_all_output_workspaces(self):
         self.assertIn('{}_{}'.format(self.raw_ws.getName(), 'properties'), mtd)
         self.assertIn('{}_{}'.format(self.raw_ws.getName(), 'refit_properties'), mtd)
 
+    def test_algorithm_works_on_specified_spectrum(self):
+        x_values = np.array([np.linspace(0, 100, 1001), np.linspace(0, 100, 1001)], dtype=float)
+        centre = np.array([[25, 75], [10, 60]], dtype=float)
+        height = np.array([[35, 20], [40, 50]], dtype=float)
+        width = np.array([[10, 5], [8, 6]], dtype=float)
+        y_values = np.array(
+            [self.gaussian(x_values[0], centre[0, 0], height[0, 0], width[0, 0]),
+             self.gaussian(x_values[1], centre[1, 0], height[1, 0], width[1, 0])])
+
+        y_values += np.array(
+            [self.gaussian(x_values[0], centre[0, 1], height[0, 1], width[0, 1]),
+             self.gaussian(x_values[1], centre[1, 1], height[1, 1], width[1, 1])])
+        background = 10 * np.ones(x_values.shape)
+        y_values += background
+
+        raw_ws = CreateWorkspace(DataX=x_values,
+                                      DataY=y_values,
```

Comment:
what if we ask for spectrum 3 on this data set?

Reply:
If you ask for spectrum 3 it would raise a ValueError as there is no spectrum 3

Reply:
could we have a test?